### PR TITLE
joystick: use long controller

### PR DESCRIPTION
**Pull request by sshane**
Puts more longcontrol behavior in the loop, such as PID gains, long control states like stopping/starting, prevent_overshoot

Bolt with joystick before this is smooth as we can modulate the accel smoothly, however the PID gains make it much more unstable which we should hit with joystick mode

### CODE REVIEW

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -616,12 +616,20 @@ def state_control(self, CS):
...
       self.LoC.reset(v_pid=CS.vEgo)
 
-    if not self.joystick_mode:
-      # accel PID loop
-      pid_accel_limits = self.CI.get_pid_accel_limits(self.CP, CS.vEgo, self.v_cruise_kph * CV.KPH_TO_MS)
+    # accel PID loop
+    if self.joystick_mode:
+      joystick_accel = 4.0 * clip(self.sm['testJoystick'].axes[0], -1, 1) if self.sm.rcv_frame['testJoystick'] > 0 else 0.0
+      long_plan = log.LongitudinalPlan(accels=[joystick_accel] * CONTROL_N,
+                                       speeds=interpolate_mpc_to_traj(LongitudinalMpc.extrapolate_lead(0, CS.vEgo, joystick_accel, 0).T[1]))
```
Comment (gast04) [inline selfdrive/controls/controlsd.py:623]:
```suggestion
                                       speeds = interpolate_mpc_to_traj(LongitudinalMpc.extrapolate_lead(0, CS.vEgo, joystick_accel, 0).T[1]))
```

Comment (sshane) [reply selfdrive/controls/controlsd.py:623]: usually keyword args have no space

Comment (gast04) [reply selfdrive/controls/controlsd.py:623]: didnt know :+1:

**Comment by sshane** [comment]
mostly feels like the old joystick now with the missing long deadzone added. not sure what the point is any more

**Pull request closed without merging**